### ISSUE TITLE
Uncaught exception while trying to connect connected app service.

### ISSUE THREAD

**Issue by prescod** [SFDO-Tooling/CumulusCI#2554]
**Describe the bug**

Uncaught exception while trying to connect connected app service.

Error should be caught and re-raised with human-readable error message (assuming it is a usage error)

**Reproduction steps**

1. cci service connect connected_app xyzyy

2.   File "/Users/pprescod/code/CumulusCI/cumulusci/cli/service.py", line 129, in callback
    if service_name in runtime.keychain.list_services()[service_type]:
KeyError: 'connected_app'

**Context**
Please include the following information at minimum.

CumulusCI version: 3.32.1 (/Users/pprescod/code/CumulusCI/cumulusci/__main__.py)
Python version: 3.9.1 (/Users/pprescod/.venvs/CCI-37/bin/python)

You have the latest version of CumulusCI.

**Comment by prescod** [comment]
Later I provoked this error:

```
cumulusci.core.exceptions.CumulusCIException: 

No DEFAULT_SERVICES.json file found at: /Users/pprescod/.cumulusci/core-sfdo-performance-tests/DEFAULT_SERVICES.json
```

**Comment by davisagli** [comment]
@prescod For the `KeyError: 'connected_app'` error, could you check if this fix helps? https://github.com/SFDO-Tooling/CumulusCI/pull/2555

For the "No DEFAULT_SERVICES.json file" error, do you have a traceback?

**Comment by davisagli** [comment]
@Br4nd0R @codexmarcel  I see a few places where the code is assuming that a project-level DEFAULT_SERVICES.json file exists and probably needs to be more defensive in case it is missing:
1. EncryptedFileProjectKeychain._remove_reference_to_alias
2. EncryptedFileProjectKeychain._rename_alias_in_default_service_file
3. EncryptedFileProjectKeychain._save_default_service

**Comment by davisagli** [comment]
@Br4nd0R @codexmarcel I also added a proposed fix for that in #2555 (make _read_default_services return an empty dict instead of raising an error if the file doesn't exist)

**Comment by prescod** [comment]
A traceback:

No DEFAULT_SERVICES.json file found at: /Users/pprescod/.cumulusci/core-sfdo-performance-tests/DEFAULT_SERVICES.json
Run this command for more information about debugging errors: cci error --help
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.1_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/Cellar/python@3.9/3.9.1_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/pprescod/code/CumulusCI/cumulusci/__main__.py", line 3, in <module>
    main()
  File "/Users/pprescod/code/CumulusCI/cumulusci/cli/cci.py", line 86, in main
    handle_exception(
  File "/Users/pprescod/code/CumulusCI/cumulusci/cli/cci.py", line 112, in handle_exception
    raise error
  File "/Users/pprescod/code/CumulusCI/cumulusci/cli/cci.py", line 78, in main
    cli(args[1:], standalone_mode=False, obj=runtime)
  File "/Users/pprescod/.venvs/CCI-37/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/pprescod/.venvs/CCI-37/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/pprescod/.venvs/CCI-37/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/pprescod/.venvs/CCI-37/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/pprescod/.venvs/CCI-37/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/pprescod/.venvs/CCI-37/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/pprescod/.venvs/CCI-37/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/pprescod/code/CumulusCI/cumulusci/cli/runtime.py", line 176, in new_func
    func(runtime, *args, **kw)
  File "/Users/pprescod/code/CumulusCI/cumulusci/cli/service.py", line 285, in service_remove
    runtime.keychain.remove_service(service_type, service_name)
  File "/Users/pprescod/code/CumulusCI/cumulusci/core/keychain/encrypted_file_project_keychain.py", line 251, in remove_service
    self._remove_reference_to_alias(
  File "/Users/pprescod/code/CumulusCI/cumulusci/core/keychain/encrypted_file_project_keychain.py", line 269, in _remove_reference_to_alias
    default_services = self._read_default_services(default_services_filepath)
  File "/Users/pprescod/code/CumulusCI/cumulusci/core/keychain/encrypted_file_project_keychain.py", line 507, in _read_default_services
    raise CumulusCIException(
cumulusci.core.exceptions.CumulusCIException: No DEFAULT_SERVICES.json file found at: /Users/pprescod/.cumulusci/core-sfdo-performance-tests/DEFAULT_SERVICES.json
